### PR TITLE
feat: DTOSS-3472 mesh client integration tests

### PR DESCRIPTION
**Pull request by MWClayson-NHS**
## Description

- Adding Integration tests
- Fixing issue with chunked files
- adding additional headers

## Context

Needed for the package to function as expected.

## Type of changes

- [X] Refactoring (non-breaking change)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [X] I have followed the code style of the project
- [X] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/NHS.Mesh.Client.IntegrationTests/MeshUncompressedMessageTests.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        string messageResponseText = System.Text.Encoding.Default.GetString(getMessageResponse.Response.FileAttachment.Content);
+        Assert.AreEqual(messageContent, messageResponseText);
+
+        //Act - Acknowledge Message
+        var acknowledgeMessageResponse = await _meshInboxService.AcknowledgeMessageByIdAsync(toMailbox, messageId!);
+
+        //Assert
+        Assert.IsTrue(acknowledgeMessageResponse.IsSuccessful);
+        Assert.AreEqual(messageId, acknowledgeMessageResponse.Response.MessageId);
+
```
Comment (andrewmorris43) [inline tests/NHS.Mesh.Client.IntegrationTests/MeshUncompressedMessageTests.cs:94]: Some extra spaces here that could be deleted.

File: tests/NHS.Mesh.Client.UnitTests/FileChunkUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+namespace NHS.MESH.Client.UnitTests;
+
+using System.Text;
+//using NHS.Mesh.Client.TestingCommon;
```
Comment (andrewmorris43) [inline tests/NHS.Mesh.Client.UnitTests/FileChunkUnitTests.cs:4]: Can this line be deleted?

**Review by andrewmorris43 — COMMENTED** [review]
Just a couple of very minor things I've spotted, but the code looks good to me.

How can I run the integration tests on my machine?

File: .editorconfig [hunk 3]
```diff
@@ -15,6 +15,10 @@ csharp_indent_braces=false
 dotnet_style_readonly_field = true
 csharp_using_directive_placement = inside_namespace
 dotnet_diagnostic.IDE0065.severity = error
+dotnet_diagnostic.IDE0005.severity = error
+
+# CS8618: Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
```
Comment (Warren-Pitterson) [inline .editorconfig:20]: is this comment needed?

File: application/DotNetMeshClient/NHS.Mesh.Client/Helpers/FileHelpers.cs [hunk 4]
```diff
@@ -72,55 +70,39 @@ public static bool IsFileTooLarge(byte[] data, long MaxLength)
...
-                await stream.CopyToAsync(gzipStream);
-            }
 
-            memoryStream.Position = 0;
+        var orderedChunks = fileChunks.OrderBy(i => i.ChunkNumber);
+        var firstChunk = orderedChunks.First();
 
-            var content = new StreamContent(memoryStream)
+        using(var memoryStream = new MemoryStream()){
+            int i = 1;
```
Comment (Warren-Pitterson) [inline application/DotNetMeshClient/NHS.Mesh.Client/Helpers/FileHelpers.cs:87]: what is i? can we give this a meaningful name please. 

it also don't get modified from here to it's implementation on line 90. The name i makes me thing it's in a loop.

File: tests/NHS.Mesh.Client.IntegrationTests/Helpers/IntegrationTestHelpers.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (Warren-Pitterson) [inline tests/NHS.Mesh.Client.IntegrationTests/Helpers/IntegrationTestHelpers.cs:1]: Is this file missing a namespace?

File: application/DotNetMeshClient/NHS.Mesh.Client/Helpers/FileHelpers.cs (hunk 4 above)
Comment (MWClayson-NHS) [reply application/DotNetMeshClient/NHS.Mesh.Client/Helpers/FileHelpers.cs:87]: Renamed this to `expectedChunkNumber`, this is used in foreach look on line 90 to check if we are not missing a chunk when trying to reassemble the file.

File: .editorconfig (hunk 3 above)
Comment (MWClayson-NHS) [reply .editorconfig:20]: I'm just removing the config line its referencing.

**Review by Warren-Pitterson — APPROVED** [review]

**Pull request merged**